library/osrm_impl.cpp

```diff
@@ -171,11 +159,11 @@ void OSRM_impl::increase_concurrent_query_count()
 }
 
 // proxy code for compilation firewall
-OSRM::OSRM(libosrm_config &lib_config) : OSRM_pimpl_(osrm::make_unique<OSRM_impl>(lib_config)) {}
+OSRM::OSRM(LibOSRMConfig lib_config) : OSRM_pimpl_(osrm::make_unique<OSRM_impl>(std::move(lib_config))) {}
```

Comment:
You need to define the special member functions in the implementation file.

See discussion over at https://github.com/mapbox/mapbox-gl-native/issues/3254#issuecomment-164055377